docs/syntax_and_semantics/assignment.md

```diff
@@ -11,45 +20,58 @@ local = 1
 
 # Assigns to a class variable
 @@class = 3
+
+# Assigns to a constant
+CONST = 4
+
+# Assigns to a setter method
+foo.method = 5
+foo[0] = 6
 ```
 
-Each of the above kinds of variables will be explained later on.
+### Method as assignment target
 
-Some syntax sugar that contains the `=` character is available:
+A method ending with an equals sign (`=`) is called a setter method. It can be used
+as the target of an assignment. The semantics of the assignment operator apply as
+a form of syntax sugar to the method call.
 
-```crystal
-local += 1  # same as: local = local + 1
+This is an example of a call to a setter method in typical method notation and with assignment operator.
+Both expressions are equivalent.
 
-# The above is valid with these operators:
-# +, -, *, /, %, |, &, ^, **, <<, >>
+```crystal
+def name=(value);end
```

Comment:
Why is this all on one line? :o

Reply:
I guess I wanted to emphasize that the implementation of the method is irrelevant.